### ISSUE TITLE
HTMLVarchar editor fields in CMS get created with default value of "1"

### ISSUE THREAD

**Issue by robingram** [silverstripe/silverstripe-framework#9358]
## Affected Version

4.4

## Description

If a field is of type `HTMLVarchar` then when a new object is created in the admin the WYSIWYG editor contains the text "1". This is because the field type is creating an editor with that default value.

https://github.com/silverstripe/silverstripe-framework/blob/4/src/ORM/FieldType/DBHTMLVarchar.php#L127

## Steps to Reproduce

Create a DO or PageType with a field of type HTMLVarchar and create a new object of that type in the CMS. Before the object is even saved the WYSIWYG editor for the field contains the character "1".

# Pull request 
* [x] https://github.com/silverstripe/silverstripe-framework/pull/9359

**Comment by robbieaverill** [comment]
Hey Rob 👋 does this also require you to add an HTMLEditorField for that varchar field?

**Comment by robingram** [comment]
Hey Robbie. Not explicitly, it happens with the auto generated field. We're looking at a work around to remove the field and explicitly add it back in without the value.

**Comment by robingram** [comment]
In fact I guess it only happens with the scaffolded editor field. Removing that and adding it back in explicitly is fine:

```
$fields->removeByName('Footer');
$fields->addFieldToTab('Root.Main', HTMLEditorField::create('Footer', 'Footer')->setRows(1));
```

**Comment by maxime-rainville** [comment]
The 1 bit appears to have been introduced by https://github.com/silverstripe/silverstripe-framework/commit/7048077184c9b5a1d2cf6fca188099998b64eb4f 11 years ago.

Presumably the "1" was meant to say the editor should only be 1 line?

**Comment by maxime-rainville** [comment]
Got a PR https://github.com/silverstripe/silverstripe-framework/pull/9359

**Comment by robingram** [comment]
Cool, thanks. I imagine that `HTMLVarchar` isn't widely used then and, on reflection, I can see why. Unless the field does something very clever to ensure that the HTML is correct the value could be truncated in the middle of an element and break the entire page layout.

I think I'll create a ticket to look at migrating these fields to HTMLText. :)

**Comment by Cheddam** [comment]
Resolved in #9359.
